### DISCUSSION TITLE
Cannot load /usr/local/apache2/modules/mod_auth_openidc.so into server: /usr/local/lib/libcjose.so.0: undefined symbol: RSA_set0_factors

### DISCUSSION THREAD

**Discussion by jiang-ziqiang1** [OpenIDC/mod_auth_openidc#849] [Q&A] [answered]
Hi，
I have compiled openidc from src "mod_auth_openidc-2.4.9.4.tar.gz" and build was successful.
CentOS Linux release 7.6
Apache 2.2.27
openssl 1.1.1m
cjose-0.6.1
jansson-2.13
hiredis-0.12.1

httpd.conf
LoadFile /usr/local/lib/libcjose.so
LoadFile /usr/local/lib/libjansson.so
LoadModule auth_openidc_module modules/mod_auth_openidc.so

<VirtualHost *:80>

    OIDCProviderMetadataURL http://10.**.0.***.:8080/auth/realms/nhn/.well-known/openid-configuration
    OIDCClientID st-enquete-dev
    OIDCClientSecret 51818f04-******.-******.-dfc6b834e119
    OIDCRedirectURI http://******.******.com/redirect_uri
    OIDCCryptoPassphrase 1234567890

    OIDCRemoteUserClaim employee_number
    OIDCAuthNHeader SSO_USER
    OIDCWhiteListedClaims employee_number preferred_username
    OIDCClaimPrefix

    <Location /*>
        AuthType auth-openidc
        Require valid-user
    </Location>

</VirtualHost>
when I start apache,show error:
httpd: Syntax error on line 414 of /usr/local/apache2/conf/httpd.conf: Cannot load /usr/local/lib/libcjose.so into server: /usr/local/lib/libcjose.so: undefined symbol: RSA_set0_factors

Please provide me suggestion.

**Comment by zandbelt** [comment, marked as answer]
it suggests that you're building against a different OpenSSL version than the one you're running with

**Reply by jiang-ziqiang1** [reply]
I changed OpenSSL version  from 1.1.1m  to 1.0.1u. and It is fixed. thank you !
